econml/cate_interpreter.py

```diff
@@ -0,0 +1,586 @@
+                 risk_seeking=False,
+                 splitter="best",
+                 max_depth=None,
+                 min_samples_split=2,
+                 min_samples_leaf=1,
+                 min_weight_fraction_leaf=0.,
+                 max_features=None,
+                 random_state=None,
+                 max_leaf_nodes=None,
+                 min_impurity_decrease=0.):
+        self.risk_level = risk_level
+        self.risk_seeking = risk_seeking
+        self.criterion = "gini"
+        self.splitter = splitter
+        self.max_depth = max_depth
+        self.min_samples_split = min_samples_split
+        self.min_samples_leaf = min_samples_leaf
+        self.min_weight_fraction_leaf = min_weight_fraction_leaf
+        self.max_features = max_features
+        self.random_state = random_state
+        self.max_leaf_nodes = max_leaf_nodes
+        self.min_impurity_decrease = min_impurity_decrease
+
+    def interpret(self, cate_estimator, X, sample_treatment_costs=None, treatment_names=None):
+        """
+        Interpret a policy based on a linear CATE estimator when applied to a set of features
+
+        Parameters
+        ----------
+        cate_estimator : :class:`.LinearCateEstimator`
+            The fitted estimator to interpret
+
+        X : array-like
+            The features against which to interpret the estimator;
+            must be compatible shape-wise with the features used to fit
+            the estimator
+
+        sample_treatment_costs : array-like, optional
+            The cost of treatment.  Can be a scalar or a variable cost with the same number of rows as ``X``
+
+        treatment_names : list of string, optional
+            The names of the two treatments
+        """
+        self.tree_model = DecisionTreeClassifier(criterion=self.criterion,
+                                                 splitter=self.splitter,
+                                                 max_depth=self.max_depth,
+                                                 min_samples_split=self.min_samples_split,
+                                                 min_weight_fraction_leaf=self.min_weight_fraction_leaf,
+                                                 max_features=self.max_features,
+                                                 random_state=self.random_state,
+                                                 max_leaf_nodes=self.max_leaf_nodes,
+                                                 min_impurity_decrease=self.min_impurity_decrease)
+        if self.risk_level is None:
+            y_pred = cate_estimator.const_marginal_effect(X)
+        elif not self.risk_seeking:
+            y_pred, _ = cate_estimator.const_marginal_effect_interval(X, alpha=self.risk_level)
+        else:
+            _, y_pred = cate_estimator.const_marginal_effect_interval(X, alpha=self.risk_level)
+
+        assert (y_pred.ndim == 1) or (y_pred.shape[1] == 1), "Only binary treatment interpretation available!"

```

Comment:
Same as above, how about multi dimensional Y?

Reply:
Also addressed this assertion.